### PR TITLE
Doc: Update stripe assignment initialization documentation.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
Adds internal documentation for stripe assignment initialization.

### CODE REVIEW

File: doc/developer-guide/cache-architecture/cache-initialization.en.rst [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+* Directory
+
+* Disk
+
+* Raw device
+
+After creating all the :cpp:class:`Span` instances, they are grouped by device
+ID to internal linked lists attached to the :cpp:member:`Store::disk`
+array[#store-disk-array]_. Spans that refer to the same directory, disk, or raw
```
Comment (persiaAziz) [inline doc/developer-guide/cache-architecture/cache-initialization.en.rst]: The output looks like `array[#store-disk-array]_` in my browser. Is this how it is supposed to look like?

File: doc/developer-guide/cache-architecture/cache-initialization.en.rst [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+For each valid span, an instance of :cpp:class:`CacheDisk` is created. This
+class is a :term:`continuation` and so can be used to perform potentially
+blocking operations on the span. The primary use of these is to be passed to
+the AIO threads as the callback when an I/O operation completes. These are then
+dispatched to AIO threads to perform :term:`storage unit` initialization. After
+all of those have completed, the resulting storage is distributed across the
+:term:`volumes <cache volume>` in :func:`cplist_reconfigure`. The
+:cpp:class:`CacheVol` instances are created at this time.
+
+Stripe Assignment
```
Comment (persiaAziz) [inline doc/developer-guide/cache-architecture/cache-initialization.en.rst]: Perhaps two/three lines describing what the stripe assignment means in terms of retrieving content will be a good?

File: doc/developer-guide/cache-architecture/cache-initialization.en.rst [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+=================
+
+:term:`Cache stripe <cache stripe>` assignment setup is done once all stripes have initialized (that
+is, all stripe header read operations have completed). There is an instance of
+:cpp:class:`CacheHostRecord` for each line in :file:`hosting.config` and one generic instance
+(:member:`Cache::hosttable`) for all cache volumes that are not explicitly assigned. If
+:file:`hosting.config` is empty then all cache volumes will be in the generic record.
+
+:func:`build_vol_hash_table` in :ts:git:`iocore/cache/Cache.cc` does the work of setting up the
```
Comment (persiaAziz) [inline doc/developer-guide/cache-architecture/cache-initialization.en.rst]: The documentation describing content retrieval from the cache can be linked here.

File: doc/developer-guide/cache-architecture/cache-initialization.en.rst (hunk 1 above)
Comment (SolidWallOfCode) [reply doc/developer-guide/cache-architecture/cache-initialization.en.rst]: Fixed.

File: doc/developer-guide/cache-architecture/cache-initialization.en.rst (hunk 2 above)
Comment (SolidWallOfCode) [reply doc/developer-guide/cache-architecture/cache-initialization.en.rst]: I put in `Cache::key_to_vol` and a reference to it at the end.

**Comment by persiaAziz** [comment]
Please squash :imp:

**Pull request merged**